docs/orchestrating-elastic-stack-applications/elasticsearch/users-and-roles.asciidoc

```diff
@@ -0,0 +1,154 @@
+:parent_page_id: elasticsearch-specification
+:page_id: users-and-roles
+ifdef::env-github[]
+****
+link:https://www.elastic.co/guide/en/cloud-on-k8s/master/k8s-{parent_page_id}.html#k8s-{page_id}[View this document on the Elastic website]
+****
+endif::[]
+[id="{p}-{page_id}"]
+= Users and roles
+
+== Default elastic user
+
+When the Elasticsearch resource is created, a default user named `elastic` is created automatically, and is assigned the `superuser` role.
+
+Its password can be retrieved in a Kubernetes secret, whose name
+is based on the Elasticsearch resource name: `<elasticsearch-name>-es-elastic-user`.
+
+For example, the password of the `elastic` user for an Elasticsearch cluster named `quickstart` can be retrieved with:
+
+[source,sh]
+----
+kubectl get secret quickstart-es-elastic-user -o=jsonpath='{.data.elastic}' | base64 --decode
+----
+
+== Creating custom users
+
+=== Native realm
+
+You can create custom users in the link:https://www.elastic.co/guide/en/elasticsearch/reference/current/native-realm.html[Elasticsearch native realm] using link:https://www.elastic.co/guide/en/elasticsearch/reference/current/security-api.html#security-user-apis[Elasticsearch user management APIs].
+
+=== File realm
+
+Custom users can also be created by providing the desired link:https://www.elastic.co/guide/en/elasticsearch/reference/current/file-realm.html[file realm content]
+in Kubernetes secrets, referenced in the Elasticsearch resource.
+
+[source,yaml]
+----
+apiVersion: elasticsearch.k8s.elastic.co/{eck_crd_version}
+kind: Elasticsearch
+metadata:
+  name: elasticsearch-sample
+spec:
+  version: {version}
+  auth:
+    fileRealm:
+    - secretName: my-filerealm-secret-1
+    - secretName: my-filerealm-secret-2
+  nodeSets:
+  - name: default
+    count: 1
+----
+
+You can reference several secrets in the Elasticsearch specification.
```

Comment:
Nit: no need for a newline between these sentences

Reply:
I tend to think it makes reading the "source" asciidoc file easier, and is not rendered in the final output.